cms-2016-simulated-datasets/code/mcm_store.py

```diff
@@ -0,0 +1,318 @@
+
+
+def mcm_downloader(dataset, mcm_dir):
+    "Query dictionary and setup script from McM database"
+
+    filepath = mcm_dir + "/dict/" + dataset.replace('/', '@') + ".json"
+    if os.path.exists(filepath) and os.stat(filepath).st_size != 0:
+        print("==> " + dataset + "\n==> Already exist. Skipping...")
+        return
+
+    #cmd = "curl -s -k https://cms-pdmv.cern.ch/mcm/public/restapi/requests/"
+    cmd = "curl -s -k https://cms-pdmv-prod.web.cern.ch/mcm/public/restapi/requests/"
+
+    mcm_dict = subprocess.run(cmd + "produces" + dataset,
+                              shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    mcm_dict_out = str(mcm_dict.stdout.decode("utf-8"))
+    prepid = None
+    if mcm_dict_out != '{"results": {}}\n' or mcm_dict_out != '{"results": {}}':
+        # get prepid from mcm/dataset
+        prepid= get_from_deep_json(json.loads(mcm_dict_out), "prepid")
+
+    if prepid == None:
+        print("Error: prepid not found in mcm for " + dataset + "\n==> Skipping dataset McM dict and script",file=sys.stderr )
+        return
+
+    outfile = mcm_dir + "/dict/" + dataset.replace('/', '@') + ".json"
+    with open(outfile, 'w') as dict_file:
+           dict_file.write(mcm_dict_out)
+
+    mcm_script = subprocess.run(cmd + "get_test/" + prepid ,
+                                shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    mcm_script_out = str(mcm_script.stdout.decode("utf-8"))
+
+    if mcm_script_out == '' or mcm_script_out[0] == '{':
+        print("[ERROR] Empty McM script (get_test) for {ds}".format(ds=dataset),
+            file=sys.stderr)
+    else:
+        outfile = mcm_dir + "/scripts/" + dataset.replace('/', '@') + ".sh"
+        with open(outfile, 'w') as dict_file:
+                dict_file.write(mcm_script_out)
+
+    ### New 2016
+    # create a directory with the dataset name under mcm_dir + "/chain"
+    # create dirs
+    if dataset.endswith('MINIAODSIM'):
+        return
+    path = mcm_dir + "/chain/" + dataset.replace('/', '@')
+    os.makedirs(path, exist_ok=True)
+
+    # command line: curl -s -k https://cms-pdmv.cern.ch/mcm/public/restapi/requests/get/<prepid> | jq .results.member_of_chain
+    # FIXME: change shell jq to deep json query
+    mcm_chain_prepid = subprocess.run(cmd + "get/" + prepid  + " | jq .results.member_of_chain",
+                                shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    mcm_chain_prepid_out = str(mcm_chain_prepid.stdout.decode("utf-8"))
+    chain_prepid = re.sub(r"[\[\]]", "", mcm_chain_prepid_out).strip()
+
+    # now query for the full chain
+    # commands line: curl -L -s -b cookies.txt https://cms-pdmv.cern.ch/mcm/restapi/chained_requests/get/<chain_prepid> | jq .results.chain
+    # FIXME: change shell jq to deep json query
+    # REQUIRES: run on command line first: auth-get-sso-cookie -u  https://cms-pdmv.cern.ch/mcm -o cookies.txt
```

Comment:
The `auth-get-sso-cookie` bit is already explained in the README file, so can be removed here for the sake of brevity?